app/models/project.rb

```diff
@@ -29,6 +29,11 @@ class Project < ApplicationRecord
 
   scope :joinable_except, ->(project_ids) { joinable.where.not(id: project_ids) }
 
+  projects_by_team = lambda do |projects, team|
+    projects.joins(:teams).preload(:tag_group).where(teams: { id: team })
```

Comment:
I think  you don't need to receive the project as a parameter, you'll chain this scope when you use `policy_scope`, so you can leave it like this:

```rb
scope :projects_joined_by_team, ->(team) { joins(:teams).preload(:tag_group).where(teams: { id: team }) }
```

and then you'll call it like this:

```rb
policy_scope(Project).projects_joined_by_team(current_team)
```

got it? Scopes are made to be chained to already existing queries instead of receiving them as a parameter.

Reply:
Also, since it's gonna be chained, we can rename the scope to `joined_by_team`, we can strip the `project` word from it.